syntax = "proto3";
service AnimeCrawler {

    rpc GetAnime (AnimeRequest) returns (AnimeResponse){}
    rpc GetStaff (StaffRequest) returns (StaffResponse){}
    rpc GetCharacters (CharactersRequest) returns (CharactersResponse){}
    rpc GetStudio (StudioRequest) returns (StudioResponse){}

}


message AnimeRequest{
    optional int32 id = 1;
}


message AnimeResponse{
    repeated Anime anime = 1;
}

message Anime {
    string url = 1;
    string id = 2;
    string name = 3;
    string name_rus = 4;
    string name_alt = 5;
    string type = 6;
    string total_episodes = 7;
    string current_episodes = 8;
    string next_episode_date = 9;
    string started = 10;
    string genres = 11;
    string score = 12;
    string rating = 13;
    string licensed_by = 14;
    repeated Studio studio = 15;
    string description = 16;
    repeated string related = 17;
    repeated Staff author = 18;
    repeated Characters main_heroes = 19;
    repeated string scenes = 20;
    repeated string videos = 21;
    repeated string similar = 22;
    string image_url = 23;
}

message StaffRequest{
    optional int32 id = 1;
}


message StaffResponse{
    repeated Staff staff = 1;
}


message Staff {
    string url = 1;
    repeated string id = 2;
    repeated string name = 3;
    repeated string name_rus = 4;
    repeated string occupation = 5;
    repeated string image_url = 6;
}


message CharactersRequest{
    optional int32 id = 1;
}


message CharactersResponse{
    repeated Characters characters = 1;
}

message Characters {
    string url = 1;
    repeated string id = 2;
    repeated string name = 3;
    repeated string name_rus = 4;
    repeated string image_url = 5;
}

message StudioRequest{
    optional int32 id = 1;
}


message StudioResponse{
    repeated Studio studio = 1;
}


message Studio {
    string url = 1;
    string id = 2;
    string name = 3;
    string image_url = 4;
}